src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -743,36 +751,33 @@ private void flush(boolean commitTranslog, boolean force, boolean waitIfOngoing)
                     } catch (Throwable e) {
                         throw new FlushFailedEngineException(shardId, e);
                     }
-                }
 
-                // We don't have to do this here; we do it defensively to make sure that even if wall clock time is misbehaving
-                // (e.g., moves backwards) we will at least still sometimes prune deleted tombstones:
-                if (engineConfig.isEnableGcDeletes()) {
-                    pruneDeletedTombstones();
                 }
-            }
 
-            // reread the last committed segment infos
-            store.incRef();
-            try (ReleasableLock _ = readLock.acquire()) {
-                ensureOpen();
-                lastCommittedSegmentInfos = store.readLastCommittedSegmentsInfo();
-            } catch (Throwable e) {
-                if (isClosed.get() == false) {
-                    logger.warn("failed to read latest segment infos on flush", e);
-                    if (Lucene.isCorruptionException(e)) {
-                        throw new FlushFailedEngineException(shardId, e);
+                // reread the last committed segment infos
+                try {
+                    ensureOpen();
```

Comment:
I think we can remove this one because the lock is aquired higher up (and we should check there).